spynnaker/pyNN/models/neural_projections/connectors/fixed_probability_connector.py

```diff
@@ -137,53 +137,41 @@ def create_synaptic_block(
     def __repr__(self):
         return "FixedProbabilityConnector({})".format(self._p_connect)
 
-    def _get_view_lo_hi(self, indexes):
-        view_lo = indexes[0]
-        view_hi = indexes[-1]
-        return view_lo, view_hi
-
     @property
     @overrides(AbstractGenerateConnectorOnMachine.gen_connector_id)
     def gen_connector_id(self):
         return ConnectorIDs.FIXED_PROBABILITY_CONNECTOR.value
 
-    @overrides(AbstractGenerateConnectorOnMachine.
-               gen_connector_params)
+    @overrides(AbstractGenerateConnectorOnMachine.gen_connector_params)
     def gen_connector_params(
-            self, pre_slices, pre_slice_index, post_slices,
-            post_slice_index, pre_vertex_slice, post_vertex_slice,
-            synapse_type, synapse_info):
+            self, pre_slices, pre_slice_index, post_slices, post_slice_index,
+            pre_vertex_slice, post_vertex_slice, synapse_type, synapse_info):
         params = []
-        pre_view_lo = 0
-        pre_view_hi = synapse_info.n_pre_neurons - 1
-        if synapse_info.prepop_is_view:
-            pre_view_lo, pre_view_hi = self._get_view_lo_hi(
-                synapse_info.pre_population._indexes)
 
-        params.extend([pre_view_lo, pre_view_hi])
+        view_range = 0, synapse_info.n_pre_neurons - 1
+        if synapse_info.prepop_is_view:
+            view_range = self._get_view_lo_hi(synapse_info.pre_population)
+        params.extend(view_range)
 
-        post_view_lo = 0
-        post_view_hi = synapse_info.n_post_neurons - 1
+        view_range = 0, synapse_info.n_post_neurons - 1
         if synapse_info.postpop_is_view:
-            post_view_lo, post_view_hi = self._get_view_lo_hi(
-                synapse_info.post_population._indexes)
-
-        params.extend([post_view_lo, post_view_hi])
+            view_range = self._get_view_lo_hi(synapse_info.post_population)
+        params.extend(view_range)
 
-        params.extend([self.__allow_self_connections])
+        params.append(self.__allow_self_connections)
```

Comment:
Is there a reason why .extend has been replaced with .append here?

Reply:
Yes. We were making a list of one element to just add a single element to an ordinary list. Which is very silly.